packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -236,14 +256,9 @@ mixin class ChangeNotifier implements Listenable {
   @override
   void addListener(VoidCallback listener) {
     assert(ChangeNotifier.debugAssertNotDisposed(this));
-    if (kFlutterMemoryAllocationsEnabled && !_creationDispatched) {
-      MemoryAllocations.instance.dispatchObjectCreated(
-        library: _flutterFoundationLibrary,
-        className: '$ChangeNotifier',
-        object: this,
-      );
-      _creationDispatched = true;
-    }
+
+    mayBeDispatchObjectCreation();
```

Comment:
Is the dart tree shaker smart enough to shake out this entire method call when kFlutterMemoryAllocationsEnabled is disabled (i.e. in release mode)?

Reply:
Same question for all other call sides.

Reply:
I created app that extends ChangeNotifier, and analyzed its content when kFlutterMemoryAllocationsEnabled is false, in two cases:

1. When method maybeDispatchObjectCreation checks kFlutterMemoryAllocationsEnabled in condition (this PR)
2. When there is no check in the method

First option did not contain MemoryAllocations, while second one contained.

So, yes, tree shaker is smart enough.

Reply:
But does it also eliminate the extra method call of calling mayBeDispatchObjectCreation or do you still pay for that unnecessarily in release mode?

Reply:
For some reasons I do not see this method in both cases. Can it be optimized to become part of constructor that invokes it?

Anyway, this PR does not change anything from perspective of this code. It just moves existing code from one place to another.

How about merging this PR to unblock investigation for existing leaks, and then researching what is happening with the method?

Reply:
Per discussion, updated invocations to be under the flag kFlutterMemoryAllocationsEnabled